tests/common.bash

```diff
@@ -45,3 +64,143 @@ waitForProcess() {
+		HYPERVISOR_VERSION=$(${HYPERVISOR_PATH} --version | head -n1)
+	fi
+	VIRTIOFSD_PATH=$(kata-runtime kata-env --json | jq -r .Hypervisor.VirtioFSDaemon)
+
+	INITRD_PATH=$(kata-runtime kata-env --json | jq -r .Initrd.Path)
+}
+
+# Checks that processes are not running
+check_processes() {
+	extract_kata_env
+
+	# Only check the kata-env if we have managed to find the kata executable...
+	if [ -x "$RUNTIME_PATH" ]; then
+		local vsock_configured=$($RUNTIME_PATH kata-env | awk '/UseVSock/ {print $3}')
+		local vsock_supported=$($RUNTIME_PATH kata-env | awk '/SupportVSock/ {print $3}')
+	else
+		local vsock_configured="false"
+		local vsock_supported="false"
+	fi
+
+	general_processes=( ${HYPERVISOR_PATH} ${SHIM_PATH} )
+
+	for i in "${general_processes[@]}"; do
+		if pgrep -f "$i"; then
+			die "Found unexpected ${i} present"
+		fi
+	done
+}
+
+# Clean environment, this function will try to remove all
+# stopped/running containers.
+clean_env()
+{
+	# If the timeout has not been set, default it to 30s
+	# Docker has a built in 10s default timeout, so make ours
+	# longer than that.
+	KATA_DOCKER_TIMEOUT=${KATA_DOCKER_TIMEOUT:-30}
+	containers_running=$(sudo timeout ${KATA_DOCKER_TIMEOUT} docker ps -q)
+
+	if [ ! -z "$containers_running" ]; then
+		# First stop all containers that are running
+		# Use kill, as the containers are generally benign, and most
+		# of the time our 'stop' request ends up doing a `kill` anyway
+		sudo timeout ${KATA_DOCKER_TIMEOUT} docker kill $containers_running
+
+		# Remove all containers
+		sudo timeout ${KATA_DOCKER_TIMEOUT} docker rm -f $(docker ps -qa)
+	fi
+}
+
+clean_env_ctr()
+{
+	local count_running="$(sudo ctr c list -q | wc -l)"
+	local remaining_attempts=10
+	declare -a running_tasks=()
+	local count_tasks=0
+	local sleep_time=1
+	local time_out=10
+
+	[ "$count_running" -eq "0" ] && return 0
+
+	readarray -t running_tasks < <(sudo ctr t list -q)
+
+	info "Wait until the containers gets removed"
+
+	for task_id in "${running_tasks[@]}"; do
+		sudo ctr t kill -a -s SIGTERM ${task_id} >/dev/null 2>&1
+		sleep 0.5
+	done
+
+	# do not stop if the command fails, it will be evaluated by waitForProcess
+	local cmd="[[ $(sudo ctr tasks list | grep -c "STOPPED") == "$count_running" ]]" || true
+
+	local res="ok"
+	waitForProcess "${time_out}" "${sleep_time}" "$cmd" || res="fail"
+
+	[ "$res" == "ok" ] || sudo systemctl restart containerd
+
+	while (( remaining_attempts > 0 )); do
+		[ "${RUNTIME}" == "runc" ] && sudo ctr tasks rm -f $(sudo ctr task list -q)
+		sudo ctr c rm $(sudo ctr c list -q) >/dev/null 2>&1
+
+		count_running="$(sudo ctr c list -q | wc -l)"
+
+		[ "$count_running" -eq 0 ] && break
+
+		remaining_attempts=$((remaining_attempts-1))
+	done
+
+	count_tasks="$(sudo ctr t list -q | wc -l)"
+
+	if (( count_tasks > 0 )); then
+		die "Can't remove running contaienrs."
+	fi
+}
+
+restart_containerd_service() {
+	restart_systemd_service_with_no_burst_limit containerd || return 1
```

Comment:
This function, from common.sh, should also be added here, no?
Otherwise calling restart_containerd_service will just fail as the function above is not defined anywhere.

Reply:
Just to be super clear, from the tests repo common.sh file.

Reply:
https://github.com/kata-containers/tests/blob/main/lib/common.bash#L272-L309

Reply:
@fidencio yeah, good catch again